src/Aspire.Hosting/LaunchProfileExtensions.cs

```diff
@@ -30,26 +30,37 @@ internal static class LaunchProfileExtensions
 
     internal static NamedLaunchProfile? GetEffectiveLaunchProfile(this ProjectResource projectResource, bool throwIfNotFound = false)
     {
-        string? launchProfileName = projectResource.SelectLaunchProfileName();
-        if (string.IsNullOrEmpty(launchProfileName))
+        try
         {
-            return null;
-        }
+            string? launchProfileName = projectResource.SelectLaunchProfileName();
+            if (string.IsNullOrEmpty(launchProfileName))
+            {
+                return null;
+            }
 
-        var profiles = projectResource.GetLaunchSettings()?.Profiles;
-        if (profiles is null)
-        {
-            return null;
-        }
+            var profiles = projectResource.GetLaunchSettings()?.Profiles;
+            if (profiles is null)
+            {
+                return null;
+            }
+
+            var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
+            if (!found && throwIfNotFound)
+            {
+                var message = string.Format(CultureInfo.InvariantCulture, Resources.LaunchSettingsFileDoesNotContainProfileExceptionMessage, launchProfileName);
+                throw new DistributedApplicationException(message);
+            }
 
-        var found = profiles.TryGetValue(launchProfileName, out var launchProfile);
-        if (!found && throwIfNotFound)
+            return launchProfile is not null ? new (launchProfileName, launchProfile) : default;
+        }
+        catch (JsonException ex)
         {
-            var message = string.Format(CultureInfo.InvariantCulture, Resources.LaunchSettingsFileDoesNotContainProfileExceptionMessage, launchProfileName);
-            throw new DistributedApplicationException(message);
+            var metadata = projectResource.GetProjectMetadata();
+            var projectFileInfo = new FileInfo(metadata.ProjectPath);
+            var launchSettingsFilePath = Path.Combine(projectFileInfo.DirectoryName!, "Properties", "launchSettings.json");
+            var message = $"Failed to get effective launch profile because of malformed JSON in '{launchSettingsFilePath}' associated with project resource '{projectResource.Name}'. See inner exception for details.'";
```

Comment:
```suggestion
            var message = $"Failed to get effective launch profile for project resource '{projectResource.Name}'. There is malformed JSON in the project's launch settings file at '{launchSettingsFilePath}'.";
```

Reply:
IMO this message is a bit better. "See inner exception for details" is implicit with any exception.